hicommonwealth/commonwealth
# RESTful routes for Topics

## Link to Issue
Closes: #5255

## Description of Changes
- Adds the `ServerTopicsController` for topics-related business logic
- Moves existing topics biz logic into controller
- Changes the Topic CRUD routes to use the following endpoints:
  - Create Topic: `POST /topics`
  - Get Topics: `GET /topics`
  - Update Topic: `PATCH /topics/:topicId`
  - Delete Topic: `DELETE /topics/:topicId`
  - Update Topic Channel (Discord): `PATCH /topics/:topicId/channels/:channelId`
  - Update Topics Order: `PUT /topics-order`

## Test Plan
- Updated existing tests
- CA test locally via web app
  - Create a new topic for a community and enable "featured in sidebar"– confirm that it shows up
  - Update the name/description/etc. of the community– confirm that it's updated correctly
  - Delete the topic– confirm that it's removed
  - Click "Order sidebar topics", reorder the topics, save– confirm that the topics show in the new order
    - Refresh and confirm that the topics still show in the new order

## Deployment Plan
N/A

## Other Considerations
N/A
